### DISCUSSION TITLE
Linux Mint 20.1 Cinnamon 64-bit - installation fail

### DISCUSSION THREAD

**Discussion by BaDaHe** [puddletag/puddletag#599] [User Support] [answered]
$ sudo apt-get install python-pyqt5 python-pyparsing python-mutagen
[sudo] password for badahe:                
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Package python-mutagen is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  python3-mutagen

E: Unable to locate package python-pyqt5
E: Package 'python-mutagen' has no installation candidate

**Comment by sandrotosi** [comment, marked as answer]
@BaDaHe where did you get those commands? please check our homepage at https://github.com/puddletag/puddletag#readme where, under the "Source Code" section, it provides the list of packages for a Debian-based OS (like Mint) to install

**Comment by BaDaHe** [comment]
Thanks for the VERY prompt reply!
Because the only worthwhile package installer in Mint is Synaptic, I
tried that first - but PuddleTag was not found. Despite knowing that I
would draw a blank, I then tried Mint's SOFTWARE MANAGER but neither
the ALL APPLICATIONS or FLATPAK buttons found PuddleTag, so then on to
the SNAP store, but in Mint 20.1 that appears to be populated with
absolutely nothing! 
I got the terminal commands that I originally used from the suggested
Debian install here:   
https://docs.puddletag.net/download.html#installing-on-ubuntu
I did try your suggestion, but that still did not work for me, and this
is what was returned.
$ sudo apt install puddletag[sudo] password for
badahe:                Reading package lists... DoneBuilding dependency
tree       Reading state information... DoneE: Unable to locate package
puddletag
RegardsBdH, Australia



-----Original Message-----From: Sandro Tosi ***@***.***>R
eply-To: puddletag/puddletag <
***@***.***>To:
puddletag/puddletag ***@***.***>Cc: BaDaHe <
BaDaHe@Zoho.com>, Mention ***@***.***>Subject: Re:
[puddletag/puddletag] Linux Mint 20.1 Cinnamon 64-bit - installation
fail (#599)Date: Wed, 05 May 2021 19:59:03 -0700

@BaDaHe where did you get those commands? please check our homepage at 
https://github.com/puddletag/puddletag#readme where, under the "Source
Code" section, it provides the list of packages for a Debian-based OS
(like Mint) to install

**Reply by sandrotosi** [reply]
the instructions in the page you mentioned are wrong, and i'll have to fix them at some point; for the time being, please try:

```
apt install python3 python3-mutagen python3-configobj python3-pyparsing python3-pyqt5 python3-pyqt5.qtsvg
```

(as reported at https://github.com/puddletag/puddletag#readme in the "Source Code" section at the bottom) and then you should be able to run puddletag from source

**Reply by quietmansoath** [reply]
Hi, so I just followed these instructions since I was having the same issue and I'm not getting it to work. Downloaded dependencies fine, cloned github fine, cd'd in.

But the last PYTHONPATH=source/ ./source/puddletag step returns
Bash: ./source/puddletag no such file or directory.

**Reply by pedrobaeza** [reply]
Directory layout changed in 7f1fa4f20f154da5e45c3619c80410fdcb94ed11.

You have to adapt that command. Being on the top level directory, it would be:

```bash
PYTHONPATH=./ ./puddletag
```

**Reply by Perkolator** [reply]
> the instructions in the page you mentioned are wrong, and i'll have to fix them at some point; for the time being, please try:
> 
> ```
> apt install python3 python3-mutagen python3-configobj python3-pyparsing python3-pyqt5 python3-pyqt5.qtsvg
> ```
> 
> (as reported at https://github.com/puddletag/puddletag#readme in the "Source Code" section at the bottom) and then you should be able to run puddletag from source

Still wrong info in the homepage. The above worked fine on Mint 20.3.

I then downloaded the newest `puddletag-2.1.1.tar.gz` from releases and the program runs fine when starting from console with `./puddletag`. But how to get it to show in start menu? The homepage says to run `python setup.py install` as root but that produces this error:

```
Traceback (most recent call last):
  File "setup.py", line 11, in <module>
    from setuptools import setup
ModuleNotFoundError: No module named 'setuptools'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "setup.py", line 13, in <module>
    from distutils.core import setup
ModuleNotFoundError: No module named 'distutils.core'
```

Also, the homepage mentions to run `desktop-file-install puddletag.desktop` as root but that doesn't do anything (visible at least), no message in the console and can't see puddletag in the startmenu.

Any help? And is creating a Flatpak, or PPA out of the question? Thanks!

**Reply by sandrotosi** [reply]
> > the instructions in the page you mentioned are wrong, and i'll have to fix them at some point; for the time being, please try:
> > ```
> > apt install python3 python3-mutagen python3-configobj python3-pyparsing python3-pyqt5 python3-pyqt5.qtsvg
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > (as reported at https://github.com/puddletag/puddletag#readme in the "Source Code" section at the bottom) and then you should be able to run puddletag from source
> 
> Still wrong info in the homepage. The above worked fine on Mint 20.3.

so you're saying that the information on the homepage worked on Mint 20.3 (which does *not* have a section specific to it in the installation section, you're welcome to contribute one) and so the information are correct?

> I then downloaded the newest `puddletag-2.1.1.tar.gz` from releases and the program runs fine when starting from console with `./puddletag`. But how to get it to show in start menu?

source distributions are not meant to replace the distro packages; contact the maintainer of puddletag in Mint, or Mint user support to know how to setup menu entries there.

> The homepage says to run `python setup.py install` as root but that produces this error:
> 
> ```
> Traceback (most recent call last):
>   File "setup.py", line 11, in <module>
>     from setuptools import setup
> ModuleNotFoundError: No module named 'setuptools'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "setup.py", line 13, in <module>
>     from distutils.core import setup
> ModuleNotFoundError: No module named 'distutils.core'
> ```

not having setuptools nor distutils makes it for a very hard life for installing anything python. i suggest to start looking into your python environment.

> 
> Also, the homepage mentions to run `desktop-file-install puddletag.desktop` as root

no it doesnt, where does the homepage say that? i expect a link to a precise line where you can find that line in the homepage.

> but that doesn't do anything (visible at least), no message in the console and can't see puddletag in the startmenu.
> 
> Any help? And is creating a Flatpak, or PPA out of the question? Thanks!

It would be much more rewarding to help you out if you didnt start this whole conversation just pointing out what you think is wrong and did not provide a single element for improvement. The volunteer nature of open source doesnt include an unpaid support for whatever problems users think they may have. I would be happy to review PRs from you to improve our documentation.

Thanks for your future contributions to puddletags.

**Reply by Perkolator** [reply]
> so you're saying that the information on the homepage worked on Mint 20.3 (which does _not_ have a section specific to it in the installation section, you're welcome to contribute one) and so the information are correct?

I merely replied that the homepage `apt-get install python-pyqt5 python-pyparsing python-mutagen` instruction is still wrong, in case you forgot to correct it. The newer `apt install python3 python3-mutagen python3-configobj python3-pyparsing python3-pyqt5 python3-pyqt5.qtsvg` is working.






> source distributions are not meant to replace the distro packages; contact the maintainer of puddletag in Mint, or Mint user support to know how to setup menu entries there.

Puddletag is not available at all. I have understood that Mint uses Ubuntu as a base.

```
$ sudo apt install puddletag
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package puddletag
```

> not having setuptools nor distutils makes it for a very hard life for installing anything python. i suggest to start looking into your python environment.

I'm not a python developer nor expert, and just have been moving to linux.

> no it doesnt, where does the homepage say that? i expect a link to a precise line where you can find that line in the homepage.

On https://docs.puddletag.net/download.html, in the "Installing from source" section it says:

> Alternatively install it by running python setup.py install as root in the unzipped directory.
>  puddletag should appear in your Multimedia (or Sounds etc.) menu. If not run desktop-file-install puddletag.desktop as root in the unzipped directory.

What's with the attitude? Why do you treat people asking help like this? What's the point of this discussion area if one can't ask for help without scolded? Jeez..
